cases/management/commands/enrich_missing_bigo.py

```
@@ -0,0 +1,613 @@
+        if not source_url:
+            return None
+
+        source_url = self._validate_url_scheme(source_url)
+        parsed = urllib.parse.urlparse(source_url)
+        guessed_name = self._sanitize_download_filename(parsed.path, source.source_id)
+        out_path = self._confined_output_path(output_dir, guessed_name)
+        try:
+            request = urllib.request.Request(
+                source_url,
+                headers={"User-Agent": "jawafdehi-bigo-enrichment/1.0"},
+            )
```

Ruff (0.15.12)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)